compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java

```diff
@@ -2291,6 +2295,9 @@ private STNode parseTypeDescriptorInternal(List<STNode> qualifiers, ParserRuleCo
                                                boolean isInConditionalExpr) {
         parseTypeDescQualifiers(qualifiers);
         STToken nextToken = peek();
+        if (isQualifiedIdentifierPredeclaredPrefix(nextToken.kind)) {
+            return parseQualifiedTypeRefOrTypeDesc(qualifiers, isInConditionalExpr);
```

Comment:
We provide recovery for the last identifier of the qualified identifier

Reply:
I see. if there's a predeclared-qualified identifier we simply parse it otherwise we stick to the normal way of parsing. 
wouldn't that make things simple in all places?

My concern is, whether it is worth doing recovery for missing identifier in  "predeclared-qualified identifier" at the cost of making the code base bit more complex.

Reply:
We agreed upon doing recovery for last identifier.